src/Executor.cpp

```diff
@@ -0,0 +1,37 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include <folly/futures/QueuedImmediateExecutor.h>
+#include "src/SubscriberBase.h"
+#include "src/mixins/ExecutorMixin.h"
+
+namespace reactivesocket {
+
+// just instantiating of the template here
```

Comment:
I don't want the template to be instantiated in every compilation unit. It is unnecessary and just adds more work for compiler and linker. That's why I declared the template to be extern. Now Executor.cpp is the only place which will instantiate the template (it could be any cpp file) so the code is generated only once.